src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -0,0 +1,508 @@
+    EmberAfStatus status                     = EMBER_ZCL_STATUS_FAILURE;
+    EmberAfStatus ReadStatus                 = EMBER_ZCL_STATUS_FAILURE;
+    EmberAfStatus WriteCoolingSetpointStatus = EMBER_ZCL_STATUS_SUCCESS, WriteHeatingSetpointStatus = EMBER_ZCL_STATUS_SUCCESS;
+
+    switch (mode)
+    {
+    case EMBER_ZCL_SETPOINT_ADJUST_MODE_HEAT_AND_COOL_SETPOINTS: {
+
+        // https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/3726
+        // Behavior is not specified if the device only contains one mode
+        // and the both mode command is sent.
+        // Implemented behavior is not perfect, but spec needs to be clear before changing the implementaion
+
+        // 4.3.7.2.1. Both
+        // [4.172] The client MAY indicate Both regardless of the server feature support. The server SHALL
+        // only adjust the setpoint that it supports and not respond with an error.
+
+        // Implementation assumes that the attribute will NOT be present if the device does not support that mode.
+
+        // In auto mode we will need to change both the heating and cooling setpoints
+        ReadStatus = GetOccupiedCoolingSetpoint(aEndpointId, &CoolingSetpoint);
+        if (ReadStatus == EMBER_ZCL_STATUS_SUCCESS)
+        {
+            CoolingSetpoint            = static_cast<int16_t>(CoolingSetpoint + amount * 10);
+            CoolingSetpoint            = EnforceCoolingSetpointLimits(CoolingSetpoint, aEndpointId);
+            WriteCoolingSetpointStatus = SetOccupiedCoolingSetpoint(aEndpointId, CoolingSetpoint);
+            if (WriteCoolingSetpointStatus != EMBER_ZCL_STATUS_SUCCESS)
+            {
+                ChipLogError(Zcl, "Error: SetOccupiedCoolingSetpoint failed!");
+            }
+        }
+
+        ReadStatus = GetOccupiedHeatingSetpoint(aEndpointId, &HeatingSetpoint);
+        if (ReadStatus == EMBER_ZCL_STATUS_SUCCESS)
+        {
+            HeatingSetpoint            = static_cast<int16_t>(HeatingSetpoint + amount * 10);
+            HeatingSetpoint            = EnforceHeatingSetpointLimits(HeatingSetpoint, aEndpointId);
+            WriteHeatingSetpointStatus = SetOccupiedHeatingSetpoint(aEndpointId, HeatingSetpoint);
+            if (WriteHeatingSetpointStatus != EMBER_ZCL_STATUS_SUCCESS)
+            {
+                ChipLogError(Zcl, "Error: SetOccupiedHeatingSetpoint failed!");
+            }
+        }
+        // https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/3801
+        // Spec behavior is not defined if one was successfull and the other not what should happen.
+        // this implementation leaves the successfull one changed but will return an error.
+        if ((WriteCoolingSetpointStatus == EMBER_ZCL_STATUS_SUCCESS) && (WriteHeatingSetpointStatus == EMBER_ZCL_STATUS_SUCCESS))
+            status = EMBER_ZCL_STATUS_SUCCESS;
+        break;
+    }
+
+    case EMBER_ZCL_SETPOINT_ADJUST_MODE_COOL_SETPOINT: {
```

Comment:
`{}` not needed here and below and other places where local variables are not declared inside the case block.

Reply:
Old habits die hard - removed not required {}